### PR TITLE
Add possibility to add a badge to a project

### PR DESCRIPTION
**Pull request by Schoaf**

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/ProjectApi.java [hunk 1]
```diff
@@ -2942,5 +2945,60 @@ public void denyAccessRequest(Object projectIdOrPath, Integer userId) throws Git
...
+   * Start the Housekeeping task for a project.
+   *
+   * <pre>
+   * <code>GitLab Endpoint: POST /projects/:id/housekeeping</code>
+   * </pre>
+   *
+   * @param projectIdOrPath the project in the form of an Integer(ID), String(path), or Project instance
+   * @throws GitLabApiException if any exception occurs
+   */
+  public void triggerHousekeeping(Object projectIdOrPath) throws GitLabApiException {
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/ProjectApi.java]: This method is defined twice causing a compile failure.

File: src/main/java/org/gitlab4j/api/ProjectApi.java [hunk 2]
```diff
@@ -2942,5 +2945,60 @@ public void denyAccessRequest(Object projectIdOrPath, Integer userId) throws Git
...
+   *
+   * <pre>
+   * <code>GitLab Endpoint: POST /projects/:id/badges</code>
+   * </pre>
+   *
+   * @param projectIdOrPath the project in the form of an Integer(ID), String(path), or Project instance
+   * @param badge the badge to add
+   * @throws GitLabApiException if any exception occurs
+   */
+  public void addBadge(Object projectIdOrPath, Badge badge) throws GitLabApiException {
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/ProjectApi.java]: This method is defined twice causing a compile failure.

File: src/main/java/org/gitlab4j/api/models/Badge.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+public class Badge {
+
+  private Integer id;
+  private String link_url;
+  private String image_url;
+  private String rendered_link_url;
+  private String rendered_image_url;
+  private String kind;
+
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/models/Badge.java]: All variable and method in this file should be camel cased.

Comment (Schoaf) [reply src/main/java/org/gitlab4j/api/models/Badge.java]: Sorry, I was not sure if I should follow the json variables from GitLab api or not. Fixed.

File: src/main/java/org/gitlab4j/api/ProjectApi.java (hunk 2 above)
Comment (Schoaf) [reply src/main/java/org/gitlab4j/api/ProjectApi.java]: Sorry, it I should have had a look at it after automatic merge resolvement.

**Comment by Schoaf** [comment]
Fixed.

**Comment by Schoaf** [comment]
Sorry, I should have created a branch. Now there is also the endpoint to get badges from the api.

**Comment by Schoaf** [comment]
argh.. cannot reopen the request. sorry. I dont know GitHub that much. Have to get used to the workflow. Didn't mean to abort the pull request.

**Comment by gmessner** [comment]
@Schoaf 
I will reopen and merge and then enhance what you have done.  It will be less time for me to fix than review the code.

**Comment by gmessner** [comment]
@Schoaf 
I have completed the Project badges API implementation and released version 4.12.11 with this functionality.

**Pull request merged**